pkg/cluster/cluster.go

```diff
@@ -34,6 +36,31 @@ func (c *Cluster) Start(ctx context.Context) (<-chan struct{}, error) {
 		return nil, errors.Wrap(err, "init cluster datastore and https")
 	}
 
+	if c.config.DisableETCD {
+		ready := make(chan struct{})
+		defer close(ready)
+
+		// try to get /db/info urls first before attempting to use join url
+		joinURL := c.config.JoinURL
+		clientURLs, _, err := etcd.ClientURLs(ctx, c.clientAccessInfo)
+		if err != nil {
+			return nil, err
+		}
+		if len(clientURLs) > 0 {
+			joinURL = clientURLs[0]
+		}
+		etcdAddress, err := url.Parse(joinURL)
+		if err != nil {
+			return nil, err
+		}
+		etcdProxy, err := etcd.NewETCDProxy(true, c.config.DataDir, "https://"+etcdAddress.Hostname()+":2379")
```

Comment:
Why are we parsing the URI, then dropping everything except for the hostname? ClientURLs should already return all the info we need in the right format:
```json
{
  "members": [
    {
      "ID": 12487901665121217000,
      "name": "centos01-d282c732",
      "peerURLs": [
        "https://10.0.1.137:2380"
      ],
      "clientURLs": [
        "https://10.0.1.137:2379"
      ]
    }
  ]
}
```

Reply:
its because we dont depend only on the clientUrls, if this is the first node to join we it will use the join url with the address of hostname+":6443", so we need to rectify that by just using the hostname and statically set the port to 2379

Reply:
OK. let me suggest an alternative then.